src/main/java/com/example/icecreamup/service/TopOrderService.java

```diff
@@ -1,31 +1,35 @@
 package com.example.icecreamup.service;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
 
 import com.example.icecreamup.dao.IProductDAO;
 import com.example.icecreamup.dto.ProductDTO;
 
-public class TopOrderService implements ITopOrderService {
-
+@Primary
+@Component
+public class TopOrderService implements ITopOrderService

```

Comment:
The methods were already in place to implement these in the same way as OrderService